src/main/java/at/ac/tuwien/kr/alpha/solver/DefaultSolver.java

```diff
@@ -224,15 +231,141 @@ private boolean learnBackjumpAddFromConflict(ConflictCause conflictCause) {
 		return true;
 	}
 
+	private boolean justifyMbtAndBacktrack() {
+		mbtAtFixpoint++;
+		// Run justification if enabled and possible.
+		if (!disableJustifications && grounder instanceof ProgramAnalyzingGrounder) {
```

Comment:
You could treat the opposite case (which is considerably simpler) first, and save one indentation for this huge chunk.